tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -231,12 +231,18 @@ static PyObject * THPVariable_detach_(PyObject* self, PyObject* args)
 static double dispatch_to_CDouble(const Tensor & self) {
   AutoNoGIL no_gil;
   AutoGPU auto_gpu(self);
+  if (self.numel() != 1) {
+    throw ValueError("only one element tensors can be converted to Python scalars");
```

Comment:
did you do this to get a nicer error message?  I.e. it's not strictly necessary, right?

Reply:
Yes, that's right